embed-code-go/README.md

```diff
@@ -1,13 +1,186 @@
-# embed-code-go
-This is the implementation of embed-code written with Go. For the further information you can check [README.md](https://github.com/SpineEventEngine/embed-code/blob/master/README.md)
+# General
+This is the implementation of embed-code written in Go. For more information, you can check the [README.md](https://github.com/SpineEventEngine/embed-code/blob/master/README.md).
 
-## Installation
+Embed-code-go provides a way to embed code snippets into markdown files. This allows developers to easily include code examples within their documentation. 
+
+Key features:
+- extracts code fragments from source files and embeds them into documentation files;
+- verifies that embedded code samples are up-to-date with the source code;
+- supports flexible configuration via command line arguments or a YAML configuration file;
+- allows embedding specific code fragments or code between start and end line patterns.
+
+# Running
+
+Embed-code-go can be run in two modes:
+1. Embedding - in this mode, all documentation files are scanned for `<embed-code>` tags, and the corresponding embeddings are performed.
+2. Checking for up-to-date - in this mode, all documentation files are checked for being up-to-date. If the check fails, an error message with the corresponding files is provided.
+
+The mode is selected by the mandatory `mode` argument. If it is set to `check`, the checking for up-to-date is performed. If it is set to `embed`, the embedding is performed.
+
+The tool can be executed as a binary file or as a Go file. In the latter case, the user must have Go [installed](#installation). The development version is `1.22.1`. The binary files are stored in the `./bin` directory.
```

Comment:
Please make the version more visible. It deserves its own section.

Maybe, a good idea would be just to provide a reference to some file, which is a real source of truth.